### PR TITLE
restart manager when middleware change

### PR DESCRIPTION
**Pull request by ldpliu**
https://issues.redhat.com/browse/ACM-7489

### CODE REVIEW

File: operator/pkg/controllers/hubofhubs/globalhub_manager.go [hunk 1]
```diff
@@ -20,12 +24,19 @@ import (
...
 	"github.com/stolostron/multicluster-global-hub/operator/pkg/renderer"
 	"github.com/stolostron/multicluster-global-hub/pkg/constants"
 	"github.com/stolostron/multicluster-global-hub/pkg/transport"
 	commonutils "github.com/stolostron/multicluster-global-hub/pkg/utils"
 )
 
+var pgConnectionCache *postgres.PostgresConnection
+var kafkaConnectionCache *kafka.KafkaConnection
+
+const managerDeploymentName = "multicluster-global-hub-manager"
```
Comment (clyang82) [inline operator/pkg/controllers/hubofhubs/globalhub_manager.go:38]: suggest to move it into constants.go file

File: operator/pkg/controllers/hubofhubs/globalhub_manager.go [hunk 2]
```diff
@@ -78,6 +89,12 @@ func (r *MulticlusterGlobalHubReconciler) reconcileManager(ctx context.Context,
 		replicas = 2
 	}
 
+	if isMiddleWareChanged(r.MiddlewareConfig) {
```
Comment (clyang82) [inline operator/pkg/controllers/hubofhubs/globalhub_manager.go:92]: middleware is a word. let us make `W` as lower case

File: operator/pkg/controllers/hubofhubs/globalhub_manager.go [hunk 3]
```diff
@@ -151,6 +168,64 @@ func (r *MulticlusterGlobalHubReconciler) reconcileManager(ctx context.Context,
 	return nil
 }
 
+func restartManagerPod(ctx context.Context, kubeClient kubernetes.Interface) error {
```
Comment (clyang82) [inline operator/pkg/controllers/hubofhubs/globalhub_manager.go:171]: restart the pod will lose the logs. can we just restart the kafkaConsumer and kafkaProducer? I remembered gorm can handle it automatically @yanmxa is that true?

File: operator/pkg/controllers/hubofhubs/globalhub_manager.go (hunk 1 above)
Comment (ldpliu) [reply operator/pkg/controllers/hubofhubs/globalhub_manager.go]: Done

File: operator/pkg/controllers/hubofhubs/globalhub_manager.go (hunk 2 above)
Comment (ldpliu) [reply operator/pkg/controllers/hubofhubs/globalhub_manager.go]: Done

File: operator/pkg/controllers/hubofhubs/globalhub_manager.go (hunk 3 above)
Comment (yanmxa) [reply operator/pkg/controllers/hubofhubs/globalhub_manager.go:169]: @clyang82 @ldpliu Sorry for not replying in time. 

- The gorm can't reload/reconnect the credential automatically for now.
- I had several ideas for propagating the middleware credentials, and before we choose the restarting manager, There are the following options we should consider(both of them don't need to restart the pod/manager):
  1. The operator calls the non-k8s API of the manager to update their credential. Then the manager rebuilds the connection to these middlewares.
      As far as I know, a lot of services provide a **refreshToken** API dedicated to updating the expired credentials. So I think we can prioritize this approach.
  2. The global hub manager watches the credentials of these middlewares, and rebuild the connections in the controller.

**Pull request closed without merging**